File: VB/DXWebApplication1/CustomReportProvider.vb
Imports DevExpress.XtraReports.Services
Imports DevExpress.XtraReports.UI
Imports System
Imports System.IO
Imports System.ServiceModel

Namespace DXWebApplication1

    Public Class CustomReportProvider
        Implements IReportProvider

        Protected Property ReportDirectory As String

        Const REPORT_FILE_EXTENSION As String = ".repx"

        Public Sub New(ByVal reportDirectory As String)
            Me.ReportDirectory = reportDirectory
        End Sub

        Public Function GetReport(ByVal id As String, ByVal context As ReportProviderContext) As XtraReport Implements IReportProvider.GetReport
            Try
                Dim reportBytes = File.ReadAllBytes(Path.Combine(ReportDirectory, id & REPORT_FILE_EXTENSION))
                Using memoryStream = New MemoryStream(reportBytes)
                    Return XtraReport.FromStream(memoryStream)
                End Using
            Catch __unusedException1__ As Exception
                Throw New FaultException(New FaultReason(String.Format("Could not find report '{0}'.", id)), New FaultCode("Server"), "GetReport")
            End Try
        End Function
    End Class
End Namespace

File: CS/DXWebApplication1/CustomReportProvider.cs
using DevExpress.XtraReports.Services;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Security.Policy;
using System.ServiceModel;
using System.Web;

namespace DXWebApplication1 {
    public class CustomReportProvider : DevExpress.XtraReports.Services.IReportProvider {
        protected string ReportDirectory { get; set; }
        const string REPORT_FILE_EXTENSION = ".repx";

        public CustomReportProvider(string reportDirectory) {
            this.ReportDirectory = reportDirectory;
        }

        public XtraReport GetReport(string id, ReportProviderContext context) {
            try {
                var reportBytes = File.ReadAllBytes(Path.Combine(ReportDirectory, id + REPORT_FILE_EXTENSION));
                using (var memoryStream = new MemoryStream(reportBytes)) {
                    return XtraReport.FromStream(memoryStream);
                }
            }
            catch (Exception) {
                throw new FaultException(new FaultReason(string.Format("Could not find report '{0}'.", id)), new FaultCode("Server"), "GetReport");
            }
        }
    }
}